.github/checkstyle/checkstyle.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+mkdir -p build/reports
+
+FILE=.github/checkstyle/checkstyle-10.3-all.jar
+if ! [[ -f "$FILE" ]]; then
+  echo "Downloading checkstyle jar"
+  curl --output .github/checkstyle/checkstyle-10.3-all.jar -LO https://github.com/checkstyle/checkstyle/releases/download/checkstyle-10.3/checkstyle-10.3-all.jar
+fi
+
+java -Dconfig_loc=.github/checkstyle -Dbasedir=$(pwd) \
+  -jar .github/checkstyle/checkstyle-10.3-all.jar \
+  -c .github/checkstyle/checkstyle.xml \
```

Comment:
I was thinking about handling two use cases
* Developers are able to run checkstyle from their laptops
* GitHub action runs it as a check.

You said that you did not manage to make it work in the Gradle plugin, correct?

Reply:
first, we need to publish this uber JAR to our AWS code-artifact, then I could try to run Main class via gradle's JavaExec task